daemon/daemon.go

```diff
@@ -1032,23 +1036,56 @@ func (daemon *Daemon) TagImage(newTag reference.Named, imageName string, force b
 	return nil
 }
 
+func writeDistributionProgress(cancelFunc func(), outStream io.Writer, progressChan <-chan xfer.Progress) {
+	sf := streamformatter.NewJSONStreamFormatter()
+	for prog := range progressChan {
+		if prog.Message != "" {
+			formatted := sf.FormatStatus(prog.ID, prog.Message)
+			if _, err := outStream.Write(formatted); err != nil {
+				cancelFunc()
```

Comment:
Not as much concerned that something gets log rather than specifically that `err` is logged somewhere. What type of error is this usually and is there a specific error it is expecting here.